platform/ios/src/MGLMapView.mm

```diff
@@ -967,6 +1000,38 @@ - (void)layoutSubviews
 /// Updates `contentInset` to reflect the current window geometry.
 - (void)adjustContentInset
 {
+    UIEdgeInsets adjustedContentInsets = UIEdgeInsetsZero;
+    UIViewController *viewController = [self rootViewController];
+    BOOL automaticallyAdjustContentInset;
+    if (@available(iOS 11.0, *))
+    {
+        adjustedContentInsets = self.safeAreaInsets;
+        
+    } else {
+        adjustedContentInsets.top = viewController.topLayoutGuide.length;
```

Comment:
Is this part is working properly also for < 11?
If it is, let's have one implementation and avoidDependency to safeArea.